client/src/components/Header/SearchBar/SearchBar.js

```diff
@@ -12,20 +12,27 @@ export default class SearchBar extends React.Component {
 
     constructor(props) {
         super(props);
+        const search = QueryString.parse(window.location.search);
         this.state = {
-            textSearchQuery: '',
-            locationLat: '',
-            locationLon: '',
-            locationAddress: ''
+            textSearchQuery: search.q || '',
+            locationLat: search.lat || '',
+            locationLon: search.lon || '',
+            locationAddress: search.address || ''
         };
     }
 
     handleTextSearchChange(textSearchQuery) {
         this.setState({textSearchQuery: textSearchQuery});
     }
 
-    handleLocationSearchSelect(lat, lon, address) {
-        this.setState({locationLat: lat, locationLon: lon, locationAddress: address});
+    async handleTextSearchClear() {
+        await this.setState({textSearchQuery: ''});
+        console.log('clear');
```

Comment:
Debug print?

Reply:
come on, I would _never_

Reply:
Please remove, it's having a huge impact on performance